akka-actor-tests/src/test/scala/akka/actor/DeathWatchSpec.scala

```diff
@@ -210,19 +210,19 @@ trait DeathWatchSpec { this: AkkaSpec with ImplicitSender with DefaultTimeout =>
     }
 
     "only notify when watching" in {
-      val subject = system.actorOf(Props[EmptyActor]())
+      val subject = system.actorOf(Props[EmptyActor])
```

Comment:
This change is related to https://github.com/ohze/akka/commit/34d57cd (not included here).

And commented here: https://github.com/akka/akka/pull/28837#discussion_r406270816

where you say:
> although that commit is need for dotty 

Do you really need to remove `()`? I did a quick tests on a tiny dotty project and is does compile fine. 

```scala
case class Props(cls: Class[_])
object Props {
  def apply[T: ClassTag](): Props = Props(implicitly[ClassTag[T]].runtimeClass)
}


val p = Props[String]()
println(p.cls)

```

```shell
[info] running Main 
class java.lang.String
```

Reply:
Hmm, I think I got why this is needed. 

It's not needed by dotty, but it will be better to remove because then users can't call it as previously `Props[EmptyActor]`. 

The problem is that if we remove we make it source incompatible. We can be sure that there are many projects out there using `Props[EmptyActor]()`.

Reply:
Maybe we are tackling this from the wrong side. The goal of the PR here, if I understand it correctly, is to make the code source compatible with the Dotty compiler. That said, we should not remove `()` at all. And we should it in the Akka code base in all places that Dotty requires it. 

The day that we provide the binaries for Dotty/Scala3, a user willing to move to Dotty/Scala3 will have to add `()` in all places they omit it before.